File: 추천50제/이분탐색/10815.py
import sys
input = sys.stdin.readline

n = int(input())
cards = list(map(int,input().split()))
m = int(input())
check = list(map(int,input().split()))
#순차탐색시 O(n) => 시간초과
#이분탐색 필요띠.
cards.sort()
answer = []
for i in check:
    start = 0
    end = n-1
    while start<=end:
        mid = (start+end)//2
        if cards[mid]< i:
            start = mid+1
        else:
            end = mid-1
            result = mid
    if cards[result] == i:
        answer.append(1)
    else:
        answer.append(0)

print(' '.join(list(map(str,answer))))

File: BOJ/10815.js
let fs = require('fs');
let input = fs.readFileSync('./test.txt').toString().split('\n');
let n = Number(input.shift());
let number_cards = input.shift().split(' ').map(Number);
let m = Number(input.shift());
let to_check = input.shift().split(' ').map(Number);
let answer = [];
number_cards.sort((a, b) => a - b);

to_check.forEach((check) => {
  let left = 0;
  let right = n - 1;
  let flag = false;
  while (left <= right) {
    let mid = Math.floor((left + right) / 2);
    if (number_cards[mid] == check) {
      flag = true;
      answer.push(1);
      break;
    } else if (check < number_cards[mid]) {
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }
  if (!flag) answer.push(0);
});
console.log(answer.join(' '));
